src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/playback/GerritMissedEventsPlaybackManager.java

```diff
@@ -0,0 +1,438 @@
+        IGerritHudsonTriggerConfig config = server.getConfig();
+
+        String events = getEventsFromEventsLogPlugin(config, buildEventsLogURL(config, lowerDate));
+
+        return createEventsFromString(events);
+    }
+
+    /**
+     * Takes a string of json events and creates a collection.
+     * @param eventsString Events in json in a string.
+     * @return collection of events.
+     */
+    private Vector<ChangeBasedEvent> createEventsFromString(String eventsString) {
+        Vector<ChangeBasedEvent> events = new Vector<ChangeBasedEvent>();
+        Scanner scanner = new Scanner(eventsString);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            logger.debug("found line: {}", line);
+            JSONObject jsonObject = JSONObject.fromObject(line);
+            GerritEvent evt = GerritJsonEventFactory.getEvent(jsonObject);
+            if (evt instanceof ChangeBasedEvent) {
+                Provider provider = new Provider();
+                provider.setName(serverName);
+                ((GerritTriggeredEvent)evt).setProvider(provider);
+                events.add((ChangeBasedEvent)evt);
+            }
+        }
+        scanner.close();
+        return events;
+    }
+
+    /**
+     *
+     * @param config Gerrit config for server.
+     * @param url URL to use.
+     * @return String of gerrit events.
+     */
+    protected String getEventsFromEventsLogPlugin(IGerritHudsonTriggerConfig config, String url) {
+        logger.info("({}) Going to GET: {}", serverName, url);
+        DefaultHttpClient httpclient = new DefaultHttpClient();
+        HttpGet httpGet = new HttpGet(url);
+        if (config.getGerritProxy() != null && !config.getGerritProxy().isEmpty()) {
+            try {
+                URL proxyUrl = new URL(config.getGerritProxy());
+                HttpHost proxy = new HttpHost(proxyUrl.getHost(), proxyUrl.getPort(), proxyUrl.getProtocol());
+                httpclient.getParams().setParameter(ConnRoutePNames.DEFAULT_PROXY, proxy);
+            } catch (MalformedURLException e) {
+                logger.error("Could not parse proxy URL, attempting without proxy.", e);
+            }
+        }
+
+        httpclient.getCredentialsProvider().setCredentials(new AuthScope(null, -1),
+                config.getHttpCredentials());
+        HttpResponse execute;
+        try {
+            execute = httpclient.execute(httpGet);
+        } catch (IOException e) {
+            logger.warn(e.getMessage(), e);
+            return "";
+        }
+
+        int statusCode = execute.getStatusLine().getStatusCode();
+        logger.info("Received status code: {} for server: {}", statusCode, serverName);
+
+        if (statusCode == HttpURLConnection.HTTP_OK) {
+            try {
+                InputStream bodyStream = execute.getEntity().getContent();
+                String body = IOUtils.toString(bodyStream, "UTF-8");
```

Comment:
I don't think you can be sure that the server returned the content in UTF-8, use the content type from the response instead.